Orange/widgets/data/oweditdomain.py

```diff
@@ -1526,14 +1528,22 @@ class TimeVariableEditor(VariableEditor):
     def __init__(self, parent=None, **kwargs):
         super().__init__(parent, **kwargs)
         form = self.layout().itemAt(0)
+        self.custom_text = ""
+        self.have_date = 0
+        self.have_time = 0
 
         self.format_cb = QComboBox()
         for item, data in [("Detect automatically", (None, 1, 1))] + list(
             Orange.data.TimeVariable.ADDITIONAL_FORMATS.items()
-        ):
+        ) + [("Custom format", (self.custom_text, self.have_time,
```

Comment:
`(self.custom_text, self.have_time, self.have_date)` will be copied into the tuple and won't be changed if the user changes them.